Gems/Atom/Bootstrap/Code/Source/BootstrapSystemComponent.cpp

```diff
@@ -514,6 +541,38 @@ namespace AZ
                 return true;
             }
 
+            void BootstrapSystemComponent::SwitchRenderPipeline(const AZ::RPI::RenderPipelineDescriptor& newRenderPipelineDesc, AZ::RPI::ViewportContextPtr viewportContext)
+            {
+                AZ::RPI::RenderPipelineDescriptor pipelineDescriptor = newRenderPipelineDesc;
+                pipelineDescriptor.m_name =
+                    AZStd::string::format("%s_%i", pipelineDescriptor.m_name.c_str(), viewportContext->GetId());
+
+                if (pipelineDescriptor.m_renderSettings.m_multisampleState.m_customPositionsCount &&
+                    !RHI::RHISystemInterface::Get()->GetDevice()->GetFeatures().m_customSamplePositions)
+                {
+                    // Disable custom sample positions because they are not supported
+                    AZ_Warning(
+                        "BootstrapSystemComponent",
+                        false,
+                        "Disabling custom sample positions for pipeline %s because they are not supported on this device",
+                        pipelineDescriptor.m_name.c_str());
+                    pipelineDescriptor.m_renderSettings.m_multisampleState.m_customPositions = {};
+                    pipelineDescriptor.m_renderSettings.m_multisampleState.m_customPositionsCount = 0;
+                }
+
+                // Create new render pipeline
+                auto oldRenderPipeline = viewportContext->GetRenderScene()->GetDefaultRenderPipeline();
+                RPI::RenderPipelinePtr newRenderPipeline = RPI::RenderPipeline::CreateRenderPipelineForWindow(
+                        pipelineDescriptor, *viewportContext->GetWindowContext().get(), AZ::RPI::ViewType::Default);
+
+                // Switch render pipeline
+                viewportContext->GetRenderScene()->RemoveRenderPipeline(oldRenderPipeline->GetId());
+                viewportContext->GetRenderScene()->AddRenderPipeline(newRenderPipeline);
```

Comment:
Is having both render pipelines loaded while switching resource intensive?

Reply:
No. The render pipelines don't cost much of memory.